src/components/grid/grid-item/grid-item.component.js

```diff
@@ -1,51 +1,13 @@
 /* eslint-disable max-len */
 import React from "react";
 import PropTypes from "prop-types";
+import propTypes from "@styled-system/prop-types";
 import GridItemStyle from "./grid-item.style";
 
 const GridItem = (props) => {
-  const {
```

Comment:
Leave this as:

```jsx
const { children, responsiveSettings, ...rest } = props;
 ```
 
and spread `{...rest}` on `GridItemStyle` rather than `{...props}`